src/viam/sdk/services/mlmodel/client.cpp

```diff
@@ -49,19 +49,21 @@ std::shared_ptr<MLModelService::named_tensor_views> MLModelServiceClient::infer(
     auto& mutable_input_data_fields = *mutable_input_data.mutable_fields();
 
     for (const auto& kv : inputs) {
-        auto ib = mutable_input_data_fields.emplace(kv.first);
-        pb::Value& value = ib.first->second;
+        pb::Value& value = mutable_input_data_fields[kv.first];
```

Comment:
Working around old protobuf. It should be safe to use `operator[]` rather than `insert` and checking the bool return here because the `InferRequest` message is newly created, and the `inputs` we are walking is uniquely keyed.